### PR TITLE
tools, README: add DM config file docs

### PR DESCRIPTION
**Pull request by CaitinChen**
Via 
https://github.com/pingcap/tidb-tools/pull/123
https://github.com/pingcap/tidb-tools/pull/132
@lilin90 @csuzhangxc @morgo PTAL. Thanks.

Because this PR contains a link that points to #763, we'll merge this PR after #763 is merged.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,6 +120,10 @@
     - [Overview](tools/data-migration-overview.md)
     - [Deployment](tools/data-migration-deployment.md)
     - [Synchronize Data](tools/data-migration-practice.md)
+    + Configure
+      - [Configuration Overview](tools/dm-configuration-file-overview.md)
+      - [Task Configuration File](tools/dm-task-configuration-file-intro.md)
+      - [Argument Description](tools/dm-task-config-argument-description.md)
```
Comment (csuzhangxc) [inline README.md]: is `tools/dm-task-config-argument-description.md` exists?

File: tools/dm-task-configuration-file-intro.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+## Global configuration
+
+### Basic information configuration
+
+```
+name: test                      # The name of the task. Should be globally unique.
+task-mode: all                  # The task mode. Can be set to `full`/`incremental`/`all`.
+is-sharding: true               # Whether it is a sharding task
+meta-schema: "dm_meta"          # The downstream database that stores the `meta` information
+remove-meta: false              # Whether to remove the `meta` information (`checkpoint` and `onlineddl`) before the
```
Comment (csuzhangxc) [inline tools/dm-task-configuration-file-intro.md]: Whether to remove the `meta` information (`checkpoint` and `onlineddl`) before starting the  synchronization task?

File: tools/dm-task-configuration-file-intro.md [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    ​    threads: 16                                 # The number of the threads mydumper dumps from the upstream database instance
+    ​    chunk-filesize: 64                          # The size of the file mydumper generates
+    ​    skip-tz-utc: true						
+    ​    extra-args: "-B test -T t1,t2 --no-locks"
+
+loaders:                                            # Configuration arguments of running Loader
+    global:
+    ​    pool-size: 16                               # The number of threads that execute mydumper SQL files concurrently in Loader
+    ​    dir: "./dumped_data"                        # The directory output by mydumper that Loader reads. Directories for
+                                                    # different tasks of the same instance are different. (mydumper outputs the
```
Comment (csuzhangxc) [inline tools/dm-task-configuration-file-intro.md]:
```suggestion
                                                    # different tasks of the same instance must be different. (mydumper outputs the 
```

File: README.md (hunk 1 above)
Comment (CaitinChen) [reply README.md]: @csuzhangxc Yes. I changed `argument-explanation` to this name.

**Comment by GregoryIan** [comment]
@morgo PTAL

File: tools/dm-configuration-file-overview.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+This document gives an overview of configuration files of DM (Data Migration). 
+
+## DM process configuration files
+
+- `inventory.ini`: The configuration file of deploying DM using DM-Ansible. You need to edit it based on your machine topology. For details, see [Edit the `inventory.ini` file to orchestrate the DM cluster](../tools/data-migration-deployment.md#step-7-edit-the-inventory.ini-file-to-orchestrate-the-dm-cluster).
+- `dm-master.toml`: The configuration file of running the DM-master process, including the topology information of the DM cluster and the corresponding relationship between the MySQL instance and DM-worker (should be one-to-one relationship). When you use DM-Ansible to deploy DM, `dm-master.toml` is generated automatically.
+- `dm-worker.toml`: The configuration file of running the DM-worker process, including the configuration information of upstream MySQL instance. When you use DM-Ansible to deploy DM, `dm-worker.toml` is generated automatically.
+
+## Configuration file and creation of DM tasks
```
Comment (lilin90) [inline tools/dm-configuration-file-overview.md]:
```suggestion
## DM synchronization task configuration
```

File: tools/dm-configuration-file-overview.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+### DM task configuration file
+
+When you use DM-Ansible to deploy DM, you can find the task configuration file example as below in `<path-to-dm-ansible>/conf`:
```
Comment (lilin90) [inline tools/dm-configuration-file-overview.md]:
```suggestion
When you use DM-Ansible to deploy DM, you can find the following task configuration file template in `<path-to-dm-ansible>/conf`:
```

File: tools/dm-configuration-file-overview.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+- `task.yaml.exmaple`: The standard configuration file of the data synchronization task (a specific task corresponds to a `task.yaml`). For the introduction of the configuration file, see [Task Configuration File](../tools/dm-task-configuration-file-intro.md).
+
+### Data synchronization task creation
+
+Taking `task.yaml.example` as an example, perform the following steps to create a data synchronization task.
```
Comment (lilin90) [inline tools/dm-configuration-file-overview.md]:
```suggestion
You can perform the following steps to create a data synchronization task based on `task.yaml.example`:
```

File: tools/dm-configuration-file-overview.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+1. Copy `task.yaml.example` as `your_task.yaml`.
+2. Modify the configurations in `your_task.yaml` based on [Task Configuration File](../tools/dm-task-configuration-file-intro.md)
```
Comment (lilin90) [inline tools/dm-configuration-file-overview.md]:
```suggestion
2. Refer to the description in the [Task Configuration File](../tools/dm-task-configuration-file-intro.md) and modify the configuration in `your_task.yaml`.
```

File: tools/dm-configuration-file-overview.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+3. Create your data synchronization task using dmctl.
+
+### Description of important concepts
```
Comment (lilin90) [inline tools/dm-configuration-file-overview.md]:
```suggestion
### Important concepts
```

File: tools/dm-configuration-file-overview.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+This section shows description of some important concepts.
+
+| Concept         | Description                                                         | Configuration File                                                     |
+| ------------ | ------------------------------------------------------------ | ------------------------------------------------------------ |
+| `instance-id`  | Specifies a MySQL / MariaDB instance (if you deploy DM using DM-ansible, `host:port` is used to construct this ID) | `mysql-instance` of `dm-master.toml`;<br> `instance-id` of `task.yaml` |
```
Comment (lilin90) [inline tools/dm-configuration-file-overview.md]:
```suggestion
| `instance-id`  | Specifies a MySQL/MariaDB instance (if you deploy DM using DM-ansible, `host:port` is used to construct this ID) | `mysql-instance` of `dm-master.toml`;<br> `instance-id` of `task.yaml` |
```

File: tools/dm-configuration-file-overview.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+| DM-worker ID | Specifies a DM-worker (from the `worker-addr` parameter of `dm-worker.toml`) | `worker-addr` of `dm-worker.toml`;<br> the `-worker` / `-w` flag of the dmctl command line |
```
Comment (lilin90) [inline tools/dm-configuration-file-overview.md]:
```suggestion
| DM-worker ID | Specifies a DM-worker (from the `worker-addr` parameter of `dm-worker.toml`) | `worker-addr` of `dm-worker.toml`;<br> the `-worker`/`-w` flag of the dmctl command line |
```

File: tools/dm-configuration-file-overview.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+> **Note:** `mysql-instance` and DM-worker should have a one-to-one relationship.
```
Comment (lilin90) [inline tools/dm-configuration-file-overview.md]:
```suggestion
> **Note:** You must keep `mysql-instance` and DM-worker at a one-to-one relationship. 
```

File: tools/dm-task-configuration-file-intro.md [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+                                                   # not allowed to set it to an ID of a MySQL instance that is not within the 
+                                                   # DM-master cluster topology.
+
+    ​    meta:                                      # The position where the binlog synchronization starts when the checkpoint of 
+                                                   # the downstream database does not exist. If the checkpoint exits, this 
+                                                   # configuration does not work. 
+    ​        binlog-name: binlog-00001
+    ​        binlog-pos: 4
+
+    ​    route-rules: ["user-route-rules-schema", "user-route-rules"]       # The name of the mapping rule between the matched
```
Comment (GregoryIan) [inline tools/dm-task-configuration-file-intro.md]: I think we can make the comment simpler, like `that route rules selected from routes above `

File: tools/dm-task-configuration-file-intro.md [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+    ​        binlog-pos: 4
+
+    ​    route-rules: ["user-route-rules-schema", "user-route-rules"]       # The name of the mapping rule between the matched 
+                                                                           # table of the upstream database instance and the 
+                                                                           # downstream database
+    ​    filter-rules: ["user-filter-1", "user-filter-2"]                   # The filter rule name of binlog events of the matched 
+                                                                           # table for the upstream database instance
+    ​    column-mapping-rules: ["instance-1"]                               # The column mapping rule name of the matched table of 
+                                                                           # the upstream database instance 
+    ​    black-white-list:  "instance"                                      # The filter rule of the black white list of the
```
Comment (GregoryIan) [inline tools/dm-task-configuration-file-intro.md]: ditto

File: tools/dm-task-configuration-file-intro.md [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+summary: This document introduces the task configuration file of Data Migration. 
+category: tools
+---
+
+# Task Configuration File of Data Migration
+
+This document introduces the task configuration file of Data Migration --
+[`task.yaml`](https://github.com/pingcap/tidb-tools/blob/docs/docs/dm/zh_CN/configuration/task.yaml), including [Global configuration](#global-configuration) and [Instance configuration](#instance-configuration).
+
+For description of configuration items, see [Task Configuration Argument Description](../tools/dm-task-config-argument-description.md).
```
Comment (lilin90) [inline tools/dm-task-configuration-file-intro.md]:
```suggestion
For description of configuration items, see [Data Migration Task Configuration Options](../tools/dm-task-config-argument-description.md).
```

File: tools/dm-task-configuration-file-intro.md [hunk 15]
```diff
@@ -0,0 +1,191 @@
+---
+title: Task Configuration File of Data Migration
```
Comment (lilin90) [inline tools/dm-task-configuration-file-intro.md]:
```suggestion
title: Data Migration Task Configuration File
```

File: tools/dm-task-configuration-file-intro.md [hunk 16]
```diff
@@ -0,0 +1,191 @@
...
+summary: This document introduces the task configuration file of Data Migration. 
+category: tools
+---
+
+# Task Configuration File of Data Migration
```
Comment (lilin90) [inline tools/dm-task-configuration-file-intro.md]:
```suggestion
# Data Migration Task Configuration File
```

Comment (lilin90) [reply tools/dm-task-configuration-file-intro.md]: To keep the title consistent with that of another config file.

File: tools/dm-task-configuration-file-intro.md [hunk 17]
```diff
@@ -0,0 +1,183 @@
...
+
+        meta:                                      # The position where the binlog synchronization starts when the checkpoint of 
+                                                   # the downstream database does not exist. If the checkpoint exits, this 
+                                                   # configuration does not work. 
+            binlog-name: binlog-00001
+            binlog-pos: 4
+
+        route-rules: ["user-route-rules-schema", "user-route-rules"]       # Routing rules selected from `routes` above
+        filter-rules: ["user-filter-1", "user-filter-2"]                   # Filter rules selected from `filters` above
+        column-mapping-rules: ["instance-1"]                               # The column mapping rule name of the matched table of
```
Comment (csuzhangxc) [inline tools/dm-task-configuration-file-intro.md]: this comment can also be:
Column mapping rules selected from `column-mappings` above

**Comment by morgo** [comment]
LGTM

File: tools/dm-task-configuration-file-intro.md [hunk 18]
```diff
@@ -0,0 +1,183 @@
...
+                                                                           # `syncer-config-name` at the same time.
+            worker-count: 32                                               # The number of threads that synchronize binlog events 
+                                                                           # concurrently in Syncer
+            batch: 2000
+            max-retry: 200
+```
+
+Refer to [Feature configuration set](#feature-configuration-set) to get the configuration details of the following options:
+
+- `route-rules`
```
Comment (lilin90) [inline tools/dm-task-configuration-file-intro.md]: To make it clearer, (for these options in the list) please specify which option the users should refer to in "Feature configuration set".  Ref: https://github.com/pingcap/tidb-tools/blob/docs/docs/dm/zh_CN/configuration/configuration.md

**Comment by CaitinChen** [comment]
@lilin90 @csuzhangxc Thanks. PTAL again.

**Pull request merged**